### PR TITLE
Isolate id_minter tests

### PR DESCRIPTION
**Pull request by kenoir**
### What is this PR trying to achieve?

Part of https://github.com/wellcometrust/platform/issues/1643

### Who is this change for?

🍶

### CODE REVIEW

File: catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/fixtures/Server.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import com.twitter.finatra.http.EmbeddedHttpServer
+import org.scalatest.Suite
+import uk.ac.wellcome.platform.idminter.{Server => AppServer}
+import uk.ac.wellcome.test.fixtures.TestWith
+import uk.ac.wellcome.test.utils.AmazonCloudWatchFlag
+
+trait Server extends AmazonCloudWatchFlag { this: Suite =>
+  def withServer[R](
+    flags: Map[String, String],
+    modifyServer: (EmbeddedHttpServer) => EmbeddedHttpServer = identity
```
Comment (rgladwell) [inline catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/fixtures/Server.scala]: +1

File: catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/fixtures/IdentifersDatabase.scala [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+case class DatabaseConfig(
+  databaseName: String,
+  tableName: String,
+  database: SQLSyntax,
+  table: SQLSyntax,
+  flags: Map[String, String],
+  session: AutoSession.type
+)
+
+trait IdentifiersDatabase
```
Comment (rgladwell) [inline catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/fixtures/IdentifersDatabase.scala]: Like this, especially the naming convention. Adding `*Fixture` seems redundant.

File: catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/fixtures/IdentifersDatabase.scala [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        tableName = tableName,
+        database = identifiersDatabase,
+        table = identifiersTable,
+        flags = flags,
+        session = session
+      )
+
+      testWith(config)
+
+    } finally {
```
Comment (rgladwell) [inline catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/fixtures/IdentifersDatabase.scala]: We should probably also close the `DBSession` here too.

File: catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/modules/IdMinterWorkerTest.scala [hunk 4]
```diff
@@ -2,64 +2,68 @@ package uk.ac.wellcome.platform.idminter.modules
...
+    with Eventually
+    with ExtendedPatience
+    with Matchers
     with MockitoSugar {
 
-  private val identifiersDao = mock[IdentifiersDao]
-  private val server = defineServer.bind[IdentifiersDao](identifiersDao)
+  it("should create the Identifiers table in MySQL upon startup") {
+    withLocalSqsQueue { queueUrl =>
+      withLocalSnsTopic { topicArn =>
```
Comment (rgladwell) [inline catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/modules/IdMinterWorkerTest.scala]: Do we ever use an SNS topic without an SQS queue in our tests?

File: catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/database/IdentifiersDaoTest.scala [hunk 5]
```diff
@@ -2,69 +2,101 @@ package uk.ac.wellcome.platform.idminter.database
...
+
 class IdentifiersDaoTest
     extends FunSpec
-    with IdentifiersMysqlLocal
+    with fixtures.IdentifiersDatabase
     with Matchers {
 
-  val identifiersDao = new IdentifiersDao(DB.connect(), identifiersTable)
+  def withIdentifiersDao[R](testWith: TestWith[IdentifiersDaoFixtures, R]) =
+    withIdentifiersDatabase[R] { dbConfig =>
```
Comment (rgladwell) [inline catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/database/IdentifiersDaoTest.scala]: I think you can ommit the type parameter here, Scala should infer it for you.

**Comment by rgladwell** [comment]
Looks good, tests much improved.

File: catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/modules/IdMinterWorkerTest.scala (hunk 4 above)
Comment (kenoir) [reply catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/modules/IdMinterWorkerTest.scala]: Yes, there seem to be a few places, however most of the `...Feature` tests will always use both so there may be utility in combining these into a single trait (though i'd prefer to do that in a future PR).

**Pull request merged**